projects/packages/scheduled-updates/tests/php/class-wpcom-rest-api-v2-endpoint-update-schedules-test.php

```diff
@@ -940,6 +909,147 @@ public function test_add_log_unauthorized() {
+		$paths   = array(
+			"a\nb",
+			'=',
+			'=',
+			' ',
+			"\ntest\t",
+			'*',
+			'ünicode',
+		);
+		$request = new WP_REST_Request( 'POST', '/wpcom/v2/update-schedules' );
+		$request->set_body_params(
+			array(
+				'plugins'  => $plugins,
+				'schedule' => $this->get_schedule( 'next Monday 8:00', 'weekly', $paths ),
+			)
+		);
+		$schedule_id = Scheduled_Updates::generate_schedule_id( $plugins );
+
+		// Successful request.
+		wp_set_current_user( $this->admin_id );
+		$result = rest_do_request( $request );
+
+		$this->assertSame( 200, $result->get_status() );
+		$this->assertSame( $schedule_id, $result->get_data() );
+
+		$option_paths = Scheduled_Updates::get_health_check_paths( $schedule_id );
+		$this->assertSame(
+			array(
+				'a_b',
+				'=',
+				'',
+				'test',
+				'*',
+				'ünicode',
+			),
+			$option_paths
+		);
+	}
+
+	/**
+	 * Test remove item with paths.
+	 *
+	 * @covers ::create_item
+	 */
+	public function test_remove_item_with_with_paths() {
+		$plugins  = array( 'gutenberg/gutenberg.php' );
+		$request  = new WP_REST_Request( 'POST', '/wpcom/v2/update-schedules' );
+		$schedule = $this->get_schedule( 'next Monday 8:00', 'weekly', array( 'a', 'b' ) );
+
+		$request->set_body_params(
+			array(
+				'plugins'  => $plugins,
+				'schedule' => $schedule,
+			)
+		);
+		$schedule_id_1 = Scheduled_Updates::generate_schedule_id( $plugins );
+
+		// Successful request.
+		wp_set_current_user( $this->admin_id );
+		$result = rest_do_request( $request );
+
+		$this->assertSame( 200, $result->get_status() );
+		$this->assertSame( $schedule_id_1, $result->get_data() );
+
+		$option_paths = Scheduled_Updates::get_health_check_paths( $schedule_id_1 );
+		$this->assertSame( array( 'a', 'b' ), $option_paths );
+
+		$plugins[]     = 'wp-test-plugin/wp-test-plugin.php';
+		$schedule_id_2 = Scheduled_Updates::generate_schedule_id( $plugins );
+		$request->set_body_params(
+			array(
+				'plugins'  => $plugins,
+				'schedule' => $this->get_schedule( 'next Monday 11:00', 'daily', array( 'c', 'd' ) ),
+			)
+		);
+
+		$result = rest_do_request( $request );
+
+		$this->assertSame( 200, $result->get_status() );
+		$this->assertSame( $schedule_id_2, $result->get_data() );
+
+		$option_paths = Scheduled_Updates::get_health_check_paths( $schedule_id_2 );
+		$this->assertSame( array( 'c', 'd' ), $option_paths );
+
+		$request = new WP_REST_Request( 'DELETE', '/wpcom/v2/update-schedules/' . $schedule_id_1 );
+		$result  = rest_do_request( $request );
+		$this->assertSame( 200, $result->get_status() );
+
+		$option_paths = Scheduled_Updates::get_health_check_paths( $schedule_id_1 );
+		$this->assertSame( array(), $option_paths );
+
+		$request = new WP_REST_Request( 'DELETE', '/wpcom/v2/update-schedules/' . $schedule_id_2 );
+		$result  = rest_do_request( $request );
+		$this->assertSame( 200, $result->get_status() );
+
+		$option_paths = Scheduled_Updates::get_health_check_paths( $schedule_id_1 );
```

Comment:
Should this be `$schedule_id_2` instead?

Reply:
You're right, fixed.